commands/checkout_test.go

```diff
@@ -0,0 +1,16 @@
+package commands
+
+import (
+	"testing"
+
+	"github.com/github/hub/Godeps/_workspace/src/github.com/bmizerany/assert"
+)
+
+func TestReplaceCheckoutParam(t *testing.T) {
+	checkoutURL := "https://github.com/github/hub/pull/12"
+	args := NewArgs([]string{"checkout", "-b", checkoutURL})
+	replaceCheckoutParam(args, checkoutURL, "jingweno", "origin/master")
+
+	cmd := args.ToCmd()
+	assert.Equal(t, "git checkout -b --track -B jingweno origin/master", cmd.String())
```

Comment:
I added the check for `-b` and `--orphan` when checking out a pull request: https://github.com/github/hub/commit/6e0d23d6199283d14ac1132d58ed3fcd9db41b26